docs/summer-of-code-ideas/summer-of-code-reports/gsoc-2020-Vasyl-Teliman.md

```diff
@@ -0,0 +1,307 @@
+# Google Summer of Code 2020 Report - Vasyl Teliman
+
+###### Personal Links:
+###### [Github](https://github.com/Vasniktel) - [Linkedin](https://www.linkedin.com/in/vasyl-teliman-775421153/)
+
+###### Proposal:
+###### [Implementation of additional transformations for spirv-fuzz](https://summerofcode.withgoogle.com/projects/#4892481772060672)
+
+###### Mentors: 
+###### Alastair Donaldson, Paul Thomson
+
+## High-level feedback
+I've decided to put this section first since it's somewhat unlikely that a potential GSoC student will read through all the information in this document :)
+
+In any way, I'm glad I've had an opportunity to participate in this project. It was a pleasure working with my mentors, doing code reviews and implementing various
+interesting features.
+
+Don't be afraid to apply even if you have very little experience in the field at hand. You will learn everything you need to know along the way.
+
+## Introduction
+This document describes what was accomplished during this project, including:
+- Project objectives
+- Implementation of those objectives
+- Potential ideas for the future
+
+## Objectives and their implementation
+A number of deliverables were determined during the early days of the project:
+- Extend the set of transformations in spirv-fuzz by implementing new ones
+- Find bugs in existing graphics drivers
+- Improve the implementation of the tooling
+
+### Extend the set of transformations in spirv-fuzz
+SPIRV-fuzz works by applying a set of transformations to the original (reference) shader to produce a modified one (variant shader) (learn more about metamorphic
+fuzzing technique in [1]). Those transformations do not change 
+the semantics of the program. Thus, differences in the output of the reference and variant shaders might be caused by bugs in the graphics driver. Hence, it is important to
```

Comment:
Strictly, [1] introduces metamorphic testing, but is not about fuzzing.  I think the two GraphicsFuzz papers - "Automated testing of graphics shader compilers" and "Metamorphic testing for (graphics) compilers" are probably the most informative sources here (though I'd still cite the original metamorphic testing report).

Reply:
Fixed in 8d484e3.